internal/observability/graph.go

```diff
@@ -93,9 +93,9 @@ func newGraph(meter *metrics.Scope, logger *zap.Logger, extract ContextExtractor
 	}
 }
 
-// begin starts a call along an edge.
-func (g *graph) begin(ctx context.Context, rpcType transport.Type, direction directionName, req *transport.Request) call {
```

Comment:
Does this need to be changed? 
Were you able to measure impact after removing zap fields from call and making call method pointers?

Reply:
+1.

Reply:
if I do zap change in another diff my stack allocated object becomes a heap allocated object. [10]zap.Fields must not be attached to call to avoid that.

Reply:
If you leave graph.begin changes untouched, what is the performance impact? 

 > [10]zap.Fields must not be attached to call to avoid that.
This change is okay and looks good.